### ISSUE TITLE
Grouped product stock status improvements

### ISSUE THREAD

**Issue by AchalJ** [woocommerce/woocommerce#20927]
I have marked a grouped product as Out of Stock. But on the single product page I am still able to add the products within the grouped product in cart.

**To Reproduce**
1. Edit any grouped product or create new one
2. Add products to it
3. Change stock status to "Out of stock"
4. Now view the product. All added products to this are still can be added to the cart.

**Expected behavior**
When we set the grouped product as Out of Stock then it should show Out of stock in instead of its collection of products.

**WordPress Environment**
<details>
```
Home URL: |   | http://testsite.local
Site URL: |   | http://testsite.local
WooCommerce version: |   | 3.4.4
Log directory writable: |   | /app/public/wp-content/uploads/wc-logs/
WordPress version: |   | 4.9.7
WordPress multisite: |   | –
WordPress memory limit: |   | 256 MB
WordPress debug mode: |   |  
WordPress cron: |   |  
Language: |   | en_US
External object cache: |  
```
</details>

**Comment by rodrigoprimo** [comment]
I was able to reproduce the issue. I agree that it is odd that you can set the stock status for grouped products, but this setting is ignored, and users can still add individual products to the cart from the grouped product page. I'm not sure what is the best solution here and there is a chance I might be missing something as I'm not too familiar with this type of product.

I can see two different approaches to solve this issue: remove the "Stock status" option from the admin screen when editing grouped products or honor this setting in the frontend and display an "Out of stock" message instead of the buttons to add the individual products to the cart.

cc @woocommerce/proton for input from the rest of the team, especially from those more familiar with the history behind grouped products.

**Comment by claudiulodro** [comment]
This is how grouped products have behaved since at least WC 2.6. 

It doesn't make sense to have the parent grouped product status control whether the child products are available or not. The child items would still be individually purchase-able on their single product page, and a product can be part of multiple different grouped products. If you need a whole grouped product out-of-stock, each item in the product should be set to out-of-stock.

My recommendation is to remove the stock status option from grouped products. Ideally we could have the stock status automatically determined and synced based on the status of its child products, but that isn't feasible without custom product tables.

**Comment by rodrigoprimo** [comment]
Thanks for your input, @claudiulodro. I agree with you and, on #20953, I changed the code to hide "Stock status" for grouped products. IMHO, this PR could go in 3.5 but I see you marked this issue for 4.0. Could you please elaborate on why you think this change is better suited for 4.0?

**Comment by claudiulodro** [comment]
The automatic stock status syncing based on child product status was the reason I had it for 4.0. That feature isn't possible without custom tables, as there isn't a feasible way to get all grouped products that contain a specific child product.

**Comment by kloon** [comment]
Changing milestone to 3.5 since we are just not making the stock status available on grouped products anymore, it had no effect anyway and might have just been an oversight on our end to not hide it.

**Comment by rodrigoprimo** [comment]
>  The automatic stock status syncing based on child product status was the reason I had it for 4.0.

Maybe this part could be left to a separate feature request issue if there is demand for that? This issue was automatically closed when #20953, but feel free to reopen it if you prefer to track automatic stock status syncing here.

**Comment by lewisself64** [comment]
I've noticed an issue when converting simple products into grouped products to do with stock status. It seems the stock status of the simple product is unchanged when it's converted so on the admin it displays "instock" or "outofstock" depending on what the product was at the time it was converted. I noticed that when creating new grouped products it seems to just default to instock so would it not be a good idea to change the stock status when converting products into grouped?
